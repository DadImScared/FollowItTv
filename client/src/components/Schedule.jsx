
import React from 'react';

const Schedule = () => (<div>schedule here</div>);

export default Schedule;
