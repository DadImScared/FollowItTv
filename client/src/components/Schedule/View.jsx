
import React from 'react';

import { withStyles } from '@material-ui/core/styles';
import Masonry from 'react-masonry-component';
import DatePicker from 'material-ui-pickers/DatePicker';

import Episode from '../Episode';
import styles from '../../styles/Schedule.css';

const View = ({
  classes,
  episodeIds,
  shows,
  episodes,
  setMasonryRef,
  expandContentCb,
  selectedDate,
  handleDateChange
}) => (
  <div className={classes.wrapper}>
    <div className={classes.datePickerWrapper}>
      <DatePicker label='Select date' autoOk showTodayButton value={selectedDate} onChange={handleDateChange} />
    </div>
    <Masonry options={{ horizontalOrder: true }} ref={setMasonryRef}>
      {
        episodeIds.length ?
          episodeIds.map((item, index) => {
            // decrease z-index per card so when one is expanded it covers the element below it before masonry adjusts
            // the layout
            return (
              <div style={{ zIndex: 500 - index }} className={classes.col} key={`${episodes[item].id}-${index}`}>
                <Episode
                  handleExpandCb={expandContentCb}
                  shows={shows}
                  item={episodes[item]}
                />
              </div>
            );
          })
          :
          <div>no results</div>
      }
    </Masonry>
  </div>
);

export default withStyles(styles)(View);
