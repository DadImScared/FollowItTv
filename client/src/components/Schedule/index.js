
import React, { Component } from 'react';

import { connect } from 'react-redux';
import moment from 'moment';

import { requestSchedule } from '../../actions/schedule';
import View from './View';

export class Schedule extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedDate: new Date()
    };
  }

  componentDidMount() {
    const date = this.getDate(this.props);
    this.props.dispatch(requestSchedule(date));
    this.dateStringToSelectedDate(date);
  }

  componentDidUpdate(prevProps) {
    const oldDate = this.getDate(prevProps);
    const newDate = this.getDate(this.props);
    if (oldDate !== newDate) {
      this.props.dispatch(requestSchedule(newDate));
      this.dateStringToSelectedDate(newDate);
    }
  }

  dateStringToSelectedDate = (date) => {
    this.setState({ selectedDate: moment(date).toDate() });
  };

  handleDateChange = (date) => {
    this.setState({ selectedDate: date.toDate() });
    this.props.history.push(`/schedule/${date.format('YYYY-MM-DD')}`);
  };

  setMasonryRef = (el) => {
    if (el) {
      this.masonry = el.masonry;
    }
  };

  getDate = (props) => {
    const { match: { params: { date = moment().format('YYYY-MM-DD') } } } = props;
    return date;
  };

  expandContentCb = () => {
    setTimeout(() => this.masonry.layout(), 275); // wait for close/open transition to finish before setting layout
  };

  render() {
    const { schedule, loading, ...other } = this.props;
    const date = this.getDate(other);
    const loadingId = `GET_SCHEDULE_${date}`;
    if (loading[loadingId]) {
      return (
        <div>loading</div>
      );
    }
    if (!schedule[date] || schedule[date] && !schedule[date].length) {
      return (
        <div>no results</div>
      );
    }
    return (
      <View
        expandContentCb={this.expandContentCb}
        setMasonryRef={this.setMasonryRef}
        episodeIds={schedule[date] || []}
        handleDateChange={this.handleDateChange}
        {...this.state}
        {...this.props}
      />
    );
  }
}

const mapStateToProps = ({ schedule, shows, episodes, loading }) => ({
  schedule,
  shows,
  episodes,
  loading
});

export default connect(mapStateToProps)(Schedule);
