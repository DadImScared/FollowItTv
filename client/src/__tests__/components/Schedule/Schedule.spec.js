
import React from 'react';

import axios from 'axios';
import moment from 'moment';
import sinon from 'sinon';
import { createShallow } from '@material-ui/core/test-utils';

import { Schedule } from '../../../components/Schedule/index';

jest.mock('axios');

describe('Schedule', () => {
  let shallow;
  let props;
  const date = '2018-03-20';
  beforeAll(() => {
    shallow = createShallow();
    props = {
      schedule: {},
      loading: {},
      dispatch: jest.fn(),
      addShows: sinon.spy(() => {}),
      addSchedule: sinon.spy(() => {}),
      addEpisodes: sinon.spy(() => {}),
      match: { params: { date } }
    };
  });

  it('should render', () => {
    const response = {
      data: [{ id: 1, name: 'episode0', show: { id: 1, name: 'ewawd' } }]
    };
    axios.get.mockImplementation(() => response);
    const wrapper = shallow(<Schedule {...props} />);
    expect(props.dispatch).toHaveBeenCalledTimes(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update selectedDate to params date on mount', () => {
    const newProps = { ...props, schedule: { [date]: [] } };
    const wrapper = shallow(<Schedule {...newProps} />);
    expect(wrapper.state().selectedDate).toEqual(moment(date).toDate());
  });
});
