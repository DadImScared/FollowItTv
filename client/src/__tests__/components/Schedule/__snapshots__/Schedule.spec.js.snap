// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule should render 1`] = `
<div>
  no results
</div>
`;
